### PR TITLE
Empty strings that should be null/not present for a blank death record

### PR DESCRIPTION
**Pull request by RobertScalfani**
This PR fixes cases where the properties of a blank death record should have `null` or no value rather than blank whitespace.

### CODE REVIEW

File: VRDR.Tests/DeathRecord_Should.cs [hunk 1]
```diff
@@ -3315,7 +3316,7 @@ public void Test_MortalityRosterBundle()
             Assert.Equal("000000000042", mortalityrosterbundle.StateLocalIdentifier1);
             Assert.Equal("100000000001", mortalityrosterbundle.StateLocalIdentifier2);
             Assert.Equal("", mortalityrosterbundle.CertificationRole["code"]); // should be empty
-            Assert.Equal("", mortalityrosterbundle.PregnancyStatusHelper); // should be missing
+            Assert.Equal(null, mortalityrosterbundle.PregnancyStatusHelper); // should be missing
```
Comment (saulakravitz) [inline VRDR.Tests/DeathRecord_Should.cs:3319]: per Github auto-comment:  should this be Assert.Null(mortality...) instead?

File: VRDR/DeathRecord_submissionProperties.cs [hunk 2]
```diff
@@ -2091,6 +2136,11 @@ public string DateOfBirth
                     Extension withinCityLimits = new Extension();
                     withinCityLimits.Url = ExtensionURL.WithinCityLimitsIndicator;
                     withinCityLimits.Value = DictToCoding(value);
+                    // Coding coding = DictToCoding(value);
```
Comment (smacadam) [inline VRDR/DeathRecord_submissionProperties.cs:2139]: Does the DictToCoding function handle setting the null value correctly?

remove commented out code if not needed

File: VRDR/DeathRecord_submissionProperties.cs [hunk 3]
```diff
@@ -2489,6 +2489,7 @@ public string EthnicityLiteral
             }
             set
             {
+                // TODO: This setter should not do anything if the value is null or an empty string
```
Comment (smacadam) [inline VRDR/DeathRecord_submissionProperties.cs]: should this TODO be addressed now or does it need some clarification?

File: VRDR/DeathRecord_submissionProperties.cs [hunk 4]
```diff
@@ -2496,7 +2556,10 @@ public string EthnicityLiteral
                 InputRaceAndEthnicityObs.Component.RemoveAll(c => c.Code.Coding[0].Code == NvssEthnicity.Literal);
                 Observation.ComponentComponent component = new Observation.ComponentComponent();
                 component.Code = new CodeableConcept(CodeSystems.ComponentCode, NvssEthnicity.Literal, NvssEthnicity.Literal, null);
-                component.Value = new FhirString(value);
+                if (!String.IsNullOrEmpty(value))
```
Comment (smacadam) [inline VRDR/DeathRecord_submissionProperties.cs:2559]: For fields like this that create an observation as part of the setter, I think the IsNullOrEmpty check should be wrapped around all of the code in the setter. Check with Saul to see what the best practice is for null values in fhir

File: VRDR/DeathRecord_submissionProperties.cs [hunk 5]
```diff
@@ -353,11 +362,15 @@ public string StateSpecific
...
+                return null;
             }
             set
             {
                 // TODO: Handle case where Composition == null (either create it or throw exception)
+                if (String.IsNullOrWhiteSpace(value))
+                {
+                    return;
+                }
                 Composition.Extension.RemoveAll(ext => ext.Url == ExtensionURL.StateSpecificField);
```
Comment (smacadam) [inline VRDR/DeathRecord_submissionProperties.cs:374]: From Saul's email, fields like State Specific that have a RemoveAll or some sort of delete that clears the field should do the delete first and then check whether the value is null and create the extension if its not null or white space.

File: VRDR/DeathRecord_submissionProperties.cs [hunk 6]
```diff
@@ -5652,15 +5652,15 @@ private void _setAgeAtDeathForCode(string code, int? value)
...
-                if (IsDictEmptyOrDefault(value) && AgeAtDeathObs == null)
-                {
-                    return;
-                }
                 if (AgeAtDeathObs == null)
                 {
                     CreateAgeAtDeathObs();
                 }
                 AgeAtDeathObs.Value.Extension.RemoveAll(ext => ext.Url == ExtensionURL.BypassEditFlag);
+                if (IsDictEmptyOrDefault(value) && AgeAtDeathObs == null)
```
Comment (smacadam) [inline VRDR/DeathRecord_submissionProperties.cs:5660]: since you always create the AgeAtDeathObs, this will never be true

**Review by smacadam — APPROVED** [review]

**Pull request merged**